SignalServiceKit/src/Messages/OWSMessageManager.m

```diff
@@ -900,14 +897,12 @@ - (TSIncomingMessage *_Nullable)handleReceivedEnvelope:(OWSSignalServiceProtosEn
     }
 
     if (thread && incomingMessage) {
+        // In case we already have a read receipt for this new message (this happens sometimes).
+        [OWSReadReceiptManager.sharedManager updateIncomingMessage:incomingMessage
+                                                       transaction:transaction];
+
         // TODO: Do this synchronously.
         dispatch_async(dispatch_get_main_queue(), ^{
-            // In case we already have a read receipt for this new message (happens sometimes).
-            OWSReadReceiptsProcessor *readReceiptsProcessor =
-                [[OWSReadReceiptsProcessor alloc] initWithIncomingMessage:incomingMessage
-                                                           storageManager:self.storageManager];
-            [readReceiptsProcessor process];
-
             [OWSDisappearingMessagesJob becomeConsistentWithConfigurationForMessage:incomingMessage
```

Comment:
It'd be nice to continue to refactor the work done in this block.

Reply:
:+1: